sklearn/linear_model/_ridge.py

```diff
@@ -2156,13 +2156,23 @@ def fit(self, X, y, sample_weight=None, score_params=None):
                     self.cv_results_[:, i] = squared_errors.ravel()
             else:
                 predictions = y - (c / G_inverse_diag)
+                # represent the predictions in the original scale
+                if predictions.ndim > 1:
+                    predictions /= sqrt_sw[:, None]
+                    unscaled_y = y / sqrt_sw[:, None]
```

Comment:
Instead of computing the `unscaled_y` here, can we store `unscaled_y` before `_preprocess` data?

```python
        unscaled_y = y
        X, y, X_offset, y_offset, X_scale = _preprocess_data(
```

Or a little higher (but with a slightly bigger diff)

```python
        X, unscaled_y = validate_data(
```

Reply:
Done.